backend/fms_core/services/container.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from django.core.exceptions import ValidationError
+from fms_core.models import Container
+
+def get_container(barcode=None):
+    container = None
+    errors = []
+    warnings = []
+
+    try:
+        container = Container.objects.get(barcode=barcode)
+    except Container.DoesNotExist:
+        errors.append(f"Could not find Container with barcode {barcode}")
+
+    return (container, errors, warnings)
+
+
+def get_or_create_container(barcode=None, kind=None, name=None, coordinates=None,
+                            container_parent=None,
+                            creation_comment=None):
+    container = None
+    errors = []
+    warnings = []
+
+    container_data = dict(
+        **(dict(location=container_parent) if container_parent else dict(location__isnull=True)),
```

Comment:
Do we really need to build a dict in such a convoluted way ?
you could do 
container_data = {} and then add each key using an if (eg : if barcode : container_data["barcode"]=barcode)
or 
keys = ["barcode", "name", "coordinates"]
values = [None, "Patate", "C01"]
dict(itertools.compress(zip(key, values), values))

Reply:
More easily readable imo (you have a direct correspondance object attribute = argument)

Reply:
well ok if you like it :)